broker-parent/broker-api/src/main/java/org/entcore/broker/api/publisher/BrokerPublisherFactory.java

```diff
@@ -0,0 +1,274 @@
+                    // Get the disabled subjects array
+                    final JsonArray disabledArray = brokerConfig.getJsonArray("disabledSubjects");
+                    if (disabledArray != null) {
+                        for (int i = 0; i < disabledArray.size(); i++) {
+                            final String pattern = disabledArray.getString(i);
+                            if (pattern != null && !pattern.isEmpty()) {
+                                matchers.add(new SubjectMatcher(pattern));
+                                log.debug("Added disabled subject pattern: " + pattern);
+                            }
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                log.error("Error loading broker disabled subjects configuration", e);
+            }
+            
+            return matchers;
+        }
+        
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            // Check for Object methods like toString(), equals(), etc.
+            if (method.getDeclaringClass() == Object.class) {
+                return method.invoke(this, args);
+            }
+            
+            // Get the BrokerPublisher annotation from method or interface
+            BrokerPublisher annotation = method.getAnnotation(BrokerPublisher.class);
+            if (annotation == null) {
+                // Check interfaces for annotation
+                Class<?>[] interfaces = method.getDeclaringClass().getInterfaces();
+                for (Class<?> iface : interfaces) {
+                    try {
+                        Method ifaceMethod = iface.getMethod(method.getName(), method.getParameterTypes());
+                        annotation = ifaceMethod.getAnnotation(BrokerPublisher.class);
+                        if (annotation != null) break;
+                    } catch (NoSuchMethodException e) {
+                        // Continue searching
+                    }
+                }
+            }
+                
+            if (annotation == null) {
+                return Future.failedFuture(
+                    new UnsupportedOperationException("Method " + method.getName() + " is not a BrokerPublisher"));
+            }
+            
+            // Replace placeholders in the subject
+            final String subject = replacePlaceholders(annotation.subject(), params);
+            
+            // Check if subject is disabled
+            if (isSubjectDisabled(subject)) {
+                log.debug("Subject is disabled, skipping publish: " + subject);
+                return Future.succeededFuture();
+            }
+            
+            // Get the message to publish (first argument)
+            final Object message = args.length > 0 ? args[0] : null;
+            
+            // Serialize the message
+            final String messageJson = message != null ? mapper.writeValueAsString(message) : null;
```

Comment:
```suggestion
            final String messageJson = message == null ? null : mapper.writeValueAsString(message);
```
c.f. https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#confusingternary